### PR TITLE
12421 legal-api - missing business info warning check returning warnings for deleted/ceased partners

### PR DESCRIPTION
**Pull request by lewischenstudio**
*Issue #:* /bcgov/entity#12421

*Description of changes:*
- added filter for cessation is None

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: legal-api/tests/unit/services/warnings/test_warnings.py [hunk 1]
```diff
@@ -27,33 +27,44 @@
...
+        ('SUCCESS', 'SP', 'FM0000001', True, 0, 1, ['registration', 'conversion'], [False, True], None, None, None),
+        ('FAIL_NO_PROPRIETOR', 'SP', 'FM0000001', True, 0, 0, ['registration'], [True], None, 'NO_PROPRIETOR', 'A proprietor is required.'),
+        ('FAIL_NO_OFFICE', 'SP', 'FM0000001', False, 1, 0, ['registration'], [True], None, 'NO_BUSINESS_OFFICE', 'A business office is required.'),
+        ('FAIL_NO_COMPLETING_PARTY', 'SP', 'FM0000001', True, 1, 0, ['registration'], [False], None, 'NO_COMPLETING_PARTY', 'A completing party is required.'),
+        ('SUCCESS', 'SP', 'FM0000001', True, 1, 0, ['registration'], [True], datetime.utcnow(), 'NO_PROPRIETOR', 'A proprietor is required.'),
+        ('SUCCESS', 'SP', 'FM0000001', True, 0, 1, ['registration'], [True], datetime.utcnow(), 'NO_PROPRIETOR', 'A proprietor is required.'),
+        ('SUCCESS', 'SP', 'FM0000001', True, 1, 0, ['registration', 'conversion'], [False, True], datetime.utcnow(), 'NO_PROPRIETOR', 'A proprietor is required.'),
+        ('SUCCESS', 'SP', 'FM0000001', True, 0, 1, ['registration', 'conversion'], [False, True], datetime.utcnow(), 'NO_PROPRIETOR', 'A proprietor is required.'),
```
Comment (argush3) [inline legal-api/tests/unit/services/warnings/test_warnings.py:43]: can we rename these tests?  maybe something like `FAIL_NO_PROPRIETOR_CESSATION_DATE`

Comment (havkonstudio) [reply legal-api/tests/unit/services/warnings/test_warnings.py]: Oops. I forgot to change them as I did for partners.

Comment (lewischenstudio) [reply legal-api/tests/unit/services/warnings/test_warnings.py]: Oops. I forgot to change them as I did for partners.

File: legal-api/tests/unit/services/warnings/test_warnings.py [hunk 2]
```diff
@@ -27,33 +27,44 @@
 
 
 @pytest.mark.parametrize(
-    'test_name, legal_type, identifier, has_office, num_persons_roles, num_org_roles, filing_types, filing_has_completing_party, expected_code, expected_msg',
+    'test_name, legal_type, identifier, has_office, num_persons_roles, num_org_roles, filing_types, filing_has_completing_party, cessation_date, expected_code, expected_msg',
```
[the remark this answers is not in the document]
Comment (argush3) [reply legal-api/tests/unit/services/warnings/test_warnings.py:30]: Actually, I was just reviewing these tests and realized that check address is mocked out to always return no errors.  I can't quite remember why that was done.

File: legal-api/tests/unit/services/warnings/business/business_checks/test_firms.py [hunk 3]
```diff
@@ -419,38 +419,51 @@ def test_check_office(session, test_name, legal_type, identifier, expected_code,
...
+        ceased_party.mailing_address.street = None
+
+    if ceased_party:
+        ceased_party.save()
+
+    result = check_parties(legal_type, business)
+
 
     if expected_code:
         assert len(result) == 1
```
Comment (argush3) [inline legal-api/tests/unit/services/warnings/business/business_checks/test_firms.py:469]: can we keep the else block here?

```
else:
        assert len(result) == 0
```

File: legal-api/src/legal_api/resources/v2/business/business.py [hunk 4]
```diff
@@ -75,7 +73,7 @@ def get_businesses(identifier: str):
     q_account = request.args.get('account')
     current_app.logger.info('account info request, for account: %s', q_account)
     if q_account and jwt.has_one_of_roles([SYSTEM_ROLE, 'account_identity']):
-        token = jwt.get_token_auth_header()
+        # token = jwt.get_token_auth_header()
```
Comment (argush3) [inline legal-api/src/legal_api/resources/v2/business/business.py:76]: please uncomment L76

**Review by argush3 — COMMENTED** [review]
looks good.  once you fix the two things i mentioned i'll approve the PR.

File: legal-api/tests/unit/services/warnings/business/business_checks/test_firms.py (hunk 3 above)
Comment (lewischenstudio) [reply legal-api/tests/unit/services/warnings/business/business_checks/test_firms.py:469]: Done.

File: legal-api/src/legal_api/resources/v2/business/business.py (hunk 4 above)
Comment (lewischenstudio) [reply legal-api/src/legal_api/resources/v2/business/business.py]: The token in the line is not being used, giving pylint error, so I’m removed this line.

**Review by argush3 — APPROVED** [review]

**Pull request merged**